trunk/NDHMS/Routing/module_channel_routing.F

```diff
@@ -250,18 +269,33 @@ subroutine SUBMUSKINGCUNGE(    &
         do while (rerror .gt. 0.01 .and. aerror .ge. mindepth .and. iter .le. maxiter)
 
           !----- lower interval  --------------------
+
+! Old Code is commented out here, no way to deal with cases when h_0 is zero in new code
+!          Tw = Bw + 2*z*h_0                    !--top width of the channel inflow
+!          if(h_0 .le. 0.0) then
+!             AREA= 0.0
+!             R = 0.0
+!             WP = 0.0
+!           else
+!            AREA = (Bw * h_0 + z * (h_0*h_0) )
+!            WP = (Bw + 2 * h_0 * sqrt(1+z*z))
+!            R   = AREA/ WP
+!           endif
+!           WPk = WP*MIN((h_0/(modK*SQRT(Bw))),1.0)  ! KINEROS2 Mod.
+! End of Old Code
            Twl = Bw + 2.0*z*h_0      !--top surface water width of the channel inflow
 
             if(h_0 .gt. bfd) then !water outside of defined channel
              AREA =  (Bw + bfd * z) * bfd
              AREAC = (TwCC * (h_0 -bfd)) !assume compound component is rect. chan, that's 3 times the Tw
              WP = (Bw + 2.0 * bfd * sqrt(1.0 + z*z))
              WPC = TwCC + (2.0 * (h_0-bfd)) !WPC is 2 times the Tw
+             WPk = WP*MIN((h_0/(modK*SQRT(Bw))),1.0)  ! KINEROS2 Mod.
              R   = (AREA + AREAC)/(WP +WPC)  ! hydraulic radius
             else
               AREA = (Bw + h_0 * z ) * h_0
               WP = (Bw + 2.0 * h_0 * sqrt(1.0 + z*z))
-
+              WPk = WP*MIN((h_0/(modK*SQRT(Bw))),1.0)  ! KINEROS2 Mod.
```

Comment:
This reduced wetted perimeter for low flows. Should we eliminate this for the compound channel, when h_0 > bfd?

Reply:
is it true that if ChannK in the Routelink is 0, this will not be activated? Or should we expect answer changes to the base.  Since modK is hard coded, I'm not seeing how this is turned on/off.